internal/logic/logic.go

```diff
@@ -120,3 +122,101 @@ func (l transactionManagementServiceLogic) NewTransaction(newTransaction model.N
 		Data:    nil,
 	}
 }
+
+func (l transactionManagementServiceLogic) DownloadTransaction(id string, cookie string) *respModel.Response {
+	transactions, _, err := l.DsSvc.Get(map[string]interface{}{"transaction_id": id}, 0, 0)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrGetTransaction),
+			Data:    nil,
+		}
+	}
+	if len(transactions) == 0 {
+		log.Error("no transaction with specified transaction_id found")
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: codes.GetErr(codes.ErrGetTransaction),
+			Data:    nil,
+		}
+	}
+	req, err := http.NewRequest("GET", l.UtilSvc.UserSvc+"/microbank/v1/user", nil)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFetchinDataUserSvc),
+			Data:    nil,
+		}
+	}
+	req.AddCookie(&http.Cookie{Name: "token", Value: cookie})
+	client := http.Client{Timeout: 2 * time.Second}
+	response, err := client.Do(req)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFetchinDataUserSvc),
+			Data:    nil,
+		}
+	}
+	if response.StatusCode != http.StatusOK {
+		log.Info("Status Not OK", response.StatusCode)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFetchinDataUserSvc),
+			Data:    nil,
+		}
+	}
+	var userResp respModel.Response
+	by, err := ioutil.ReadAll(response.Body)
+	if err != nil {
```

Comment:
Looks like this flows is not covered. Can this be added?
https://app.codecov.io/gh/vatsal278/TransactionManagementService/pull/4#ebea564d94e0d939ccfa29b6ab98b11a-R174